### PR TITLE
allow filtering models during load

### PR DESCRIPTION
**Pull request by snopoke**
After seeing a few errors come up with the ICDS staging migration I think it could be really helpful to re-look at the dump / load process to try and make it more 'resumable'.

This was an initial attempt to allow importing data by doc type. Unfortunately due to how the data is dumped (in large gzip files) it's not possible to do efficient filtering.

I had considered dumping the data into a zip file with one entry per doc (`doc_type/doc_id.json`) but I think that might result in too many entries in the zip file for large domains.

A related option would be to dump each doc type into it's own gzip file and then add that to the final zip file.

If a failure during load does happen there is still a need to delete the docs that have been loaded - at least for the doc type where the error occurred (currently you have to delete everything and start fresh).

### CODE REVIEW

File: corehq/apps/dump_reload/management/commands/load_domain_data.py [hunk 1]
```diff
@@ -80,9 +85,10 @@ def extract_dump_archive(self, dump_file_path):
                 "Extracted dump already exists at {}. Delete it or use --use-extracted".format(target_dir))
         return target_dir
 
-    def _load_data(self, loader_class, extracted_dump_path):
+    def _load_data(self, loader_class, extracted_dump_path, object_filter):
         try:
-            return loader_class(self.stdout, self.stderr).load_from_file(extracted_dump_path, self.force)
+            loader = loader_class(object_filter, self.stdout, self.stderr)
+            loader.load_from_file(extracted_dump_path, self.force)
```
Comment (esoergel) [inline corehq/apps/dump_reload/management/commands/load_domain_data.py:91]: I think this value needs to be `return`ed

**Review by esoergel — CHANGES_REQUESTED** [review]
Looks like a missing `return`, otherwise good to go.

> If a failure during load does happen there is still a need to delete the docs that have been loaded - at least for the doc type where the error occurred (currently you have to delete everything and start fresh).

Do you know of an existing util for this, or were you imagining it being done manually for now?

File: corehq/apps/dump_reload/management/commands/load_domain_data.py [hunk 2]
```diff
@@ -47,13 +51,14 @@ def handle(self, dump_file_path, **options):
...
         loaders = options.get('loaders')
+        object_filter = options.get('object_filter')
         if loaders:
             loaders = [loader for loader in LOADERS if loader.slug in loaders]
         else:
             loaders = LOADERS
 
         for loader in loaders:
-            loader_total_object_count, loader_model_counts = self._load_data(loader, extracted_dir)
+            loader_total_object_count, loader_model_counts = self._load_data(object_filter, loader, extracted_dir)
```
Comment (kaapstorm) [inline corehq/apps/dump_reload/management/commands/load_domain_data.py:61]: `object_filter` is the third parameter of `_load_data()`, but this is passing the value in the first parameter.

**Review by kaapstorm — COMMENTED** [review]
Cool.

File: corehq/apps/dump_reload/management/commands/load_domain_data.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/dump_reload/management/commands/load_domain_data.py]: :+1: 014a6dc30ac

**Review by kaapstorm — APPROVED** [review]

**Review by esoergel — APPROVED** [review]

**Pull request merged**